fairseq/sequence_generator.py

```diff
@@ -269,6 +269,7 @@ def _generate(
 
         reorder_state: Optional[Tensor] = None
         batch_idxs: Optional[Tensor] = None
+        original_batch_idxs = sample.get("id", torch.arange(0, bsz))
```

Comment:
This line seems to be causing the test failure

Reply:
I understand! But apparently the `id` field can be of different types as I got:
```
Variable 'original_batch_idxs' previously has type Optional[Tensor] but is now being assigned to a value of type Dict[str, Tensor]
```
This is weird as the filed `id` is supposed to contain an unidimensional tensor of positive integer ids.

Reply:
@myleott I changed a bit the type check for `original_batch_idxs`. I also had to change the signature for `search.Search.step` adding 2 optional parameters as with TorchScript it seems I cannot have conditional computation based on the type of a class. Not it should be ok. It passed the tests.